### PR TITLE
Update Travis infrastructure and buildbot compiler

### PR DESCRIPTION
**Pull request by sfan5**
Reasons why this should be in 5.2:
- it doesn't hurt
- the gitlab daily build pipeline currently produces [broken builds](https://forum.minetest.net/viewtopic.php?p=368547#p368547)

<b>DONT SQUASH</b>

### CODE REVIEW

**Comment by sfan5** [comment]
Question: Should clang-tidy builds be disabled?
They currently take 23 min 56 sec and I don't think anyone ever looks at the output. (which mostly consists of `v3s16`/`v3f` const-ref warnings anyway)

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -239,13 +239,13 @@ build:fedora-24:
...
 
 .generic_win_template: &generic_win_template
-  image: ubuntu:xenial
+  image: ubuntu:bionic
   before_script:
     - apt-get update -y
-    - apt-get install -y p7zip-full wget unzip git cmake gettext
-    - wget http://minetest.kitsunemimi.pw/mingw-w64-${WIN_ARCH}_7.1.1_ubuntu14.04.7z -O mingw.7z > /dev/null
+    - apt-get install -y wget unzip git cmake gettext
+    - wget -q http://minetest.kitsunemimi.pw/mingw-w64-${WIN_ARCH}_9.2.0_ubuntu18.04.tar.xz -O mingw.tar.xz
```
Comment (appgurueu) [inline .gitlab-ci.yml]: Can't we find another option to get those files? HTTP is insecure, so probably one could sneak in corrupted dependency files?

**Comment by paramat** [comment]
> Should clang-tidy builds be disabled?

Yes please. For every PR the output complains about the same many hundreds of things which will not be changed for a long time or ever. So i take no notice of it, anything useful is buried and impossible to find.
I have also noticed how unbearably slow build checks are, which just results in a higher chance of merging before checks have finished.

**Comment by appgurueu** [comment]
> For every PR the output complains about the same many hundreds of things which will not be changed for a long time or ever. So i take no notice of it, anything useful is buried and impossible to find.

Indeed. But couldn't it be [configured](https://clang.llvm.org/extra/clang-tidy/) so that it doesn't warn about those things? Or if (some of) the warnings are valid, then shouldn't those things be fixed?

> I have also noticed how unbearably slow build checks are

They are incredibly slow. I wonder why; using version control it should be able to run faster, as there is no need to recompile everything? Probably this could also be improved (although I don't know the best way to save state, prob. caching object files or the like?)

**Comment by sfan5** [comment]
> Indeed. But couldn't it be configured so that it doesn't warn about those things?

I didn't expect this, but it looks like the most common [warning](https://clang.llvm.org/extra/clang-tidy/checks/performance-unnecessary-value-param.html) can indeed be configured not to warn on passing `v3s16` or `v3f` by value.

**Comment by SmallJoker** [comment]
I did not test these changes, but the output looks good so far. Win32/64 builds from the GitLab pipeline are handy for occasional testing, thus I'm glad to see this being fixed.

**Comment by paramat** [comment]
> it looks like the most common warning can indeed be configured not to warn on passing v3s16 or v3f by value.

Configuration seems good to me then.

**Pull request merged**